test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/AddSecretsManagerSubscriptionCommandTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.StaticStore;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Core.Utilities;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using NSubstitute.ReceivedExtensions;
+using Xunit;
+
+namespace Bit.Core.Test.OrganizationFeatures.OrganizationSubscriptionUpdate;
+[SutProviderCustomize]
+public class AddSecretsManagerSubscriptionCommandTests
+{
+    [Theory]
+    [BitAutoData(PlanType.TeamsAnnually)]
+    [BitAutoData(PlanType.TeamsMonthly)]
+    [BitAutoData(PlanType.EnterpriseAnnually)]
+    [BitAutoData(PlanType.EnterpriseMonthly)]
+    public async Task SignUpAsync_ReturnsSuccessAndClientSecret_WhenOrganizationAndPlanExist(PlanType planType,
+        SutProvider<AddSecretsManagerSubscriptionCommand> sutProvider,
+        int additionalServiceAccounts,
+        int additionalSmSeats,
+        Organization organization)
+    {
+        organization.PlanType = planType;
+        var signup = new OrganizationUpgrade
+        {
+            UseSecretsManager = true,
+            AdditionalSmSeats = additionalSmSeats,
+            AdditionalServiceAccounts = additionalServiceAccounts,
+            AdditionalSeats = organization.Seats.GetValueOrDefault(),
+        };
+
+        var plan = StaticStore.SecretManagerPlans.FirstOrDefault(p => p.Type == organization.PlanType);
+
+        var result = await sutProvider.Sut.SignUpAsync(organization, additionalSmSeats, additionalServiceAccounts);
+
+        sutProvider.GetDependency<IOrganizationService>().Received(1)
+            .ValidateSecretsManagerPlan(plan, Arg.Is<OrganizationUpgrade>(c =>
+                c.UseSecretsManager == signup.UseSecretsManager &&
+                c.AdditionalSmSeats == signup.AdditionalSmSeats &&
+                c.AdditionalServiceAccounts == signup.AdditionalServiceAccounts &&
+                c.AdditionalSeats == organization.Seats.GetValueOrDefault()));
```

Comment:
The `signup` object isn't necessary, you can just compare use the argument values for comparison (e.g. `additionalServiceAccounts`).

Reply:
resolved